### PR TITLE
Harvester cloud provider enhancement

### PR DESCRIPTION
**Pull request by yaocw2020**
- Integrate kube-vip with the harvester cloud provider

Related issue:
https://github.com/harvester/harvester/issues/1762
https://github.com/harvester/harvester/issues/1580

design document:
https://github.com/harvester/harvester/pull/1913

### CODE REVIEW

**Review by joshmeranda — CHANGES_REQUESTED** [review]
Other than potentially bumping the kube-vip version, this looks good to me :+1:

**Review by w13915984028 — APPROVED** [review]
LGTM, thanks.

**Review by chrisho — APPROVED** [review]
LGTM, thanks.

**Review by futuretea — APPROVED** [review]
LGTM, Thanks

File: charts/harvester-cloud-provider/Chart.yaml [hunk 1]
```diff
@@ -43,3 +43,8 @@ annotations:
 
 maintainers:
   - name: harvester
+
+dependencies:
+  - name: kube-vip
+    version: 0.4.2
```
[the remark this answers is not in the document]
Comment (yaocw2020) [reply charts/harvester-cloud-provider/Chart.yaml:49]: Bump kube-vip image v0.6.0 but still use helm chart 0.4.2.

File: charts/harvester-cloud-provider/values.yaml [hunk 2]
```diff
@@ -69,3 +69,21 @@ global:
   cattle:
     systemDefaultRegistry: ""
     clusterName: ""
+
+kube-vip:
+  enabled: true
+  image:
+    repository: ghcr.io/kube-vip/kube-vip-iptables
```
Comment (guangbochen) [inline charts/harvester-cloud-provider/values.yaml:76]: can we please help to check the required image for air-gap env and private registry config. https://github.com/rancher/rke2/blob/master/scripts/build-images#L86

File: charts/harvester-cloud-provider/dependency_charts/kube-vip/templates/daemonset.yaml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+          {{- end }}
+        {{- with .Values.env }}
+          {{- range $k, $v := . }}
+            {{- $name := $k }}
+            {{- $value := $v }}
+          - name: {{ quote $name }}
+            value: {{ quote $value }}
+          {{- end }}
+        {{- end }}
+        image: {{ template "system_default_registry" . }}{{ .Values.image.repository }}:{{ .Values.image.tag | default .Chart.AppVersion }}
```
Comment (guangbochen) [inline charts/harvester-cloud-provider/dependency_charts/kube-vip/templates/daemonset.yaml:31]: need to define the template per https://github.com/rancher/rke2-charts/blob/main-source/packages/rke2-canal/charts/templates/_helpers.tpl#L1

File: charts/harvester-cloud-provider/values.yaml (hunk 2 above)
Comment (yaocw2020) [reply charts/harvester-cloud-provider/values.yaml]: Done

**Review by guangbochen — APPROVED** [review]
LGTM, thanks.

**Pull request merged**